pkg/services/ngalert/api/authorization.go

```diff
@@ -2,9 +2,130 @@ package api
+
+	// Alerting instances (+silences) actions
+	ActionAlertingInstanceCreate = "alert.instance:create"
+	ActionAlertingInstanceUpdate = "alert.instance:update"
+	ActionAlertingInstanceRead   = "alert.instance:read"
+
+	// Alerting Notification policies actions
+	ActionAlertingNotificationsCreate = "alert.notifications:create"
+	ActionAlertingNotificationsRead   = "alert.notifications:read"
+	ActionAlertingNotificationsUpdate = "alert.notifications:update"
+	ActionAlertingNotificationsDelete = "alert.notifications:delete"
+)
+
+var (
+	alertingReader = ac.FixedRolePrefix + "alerting:reader"
+	alertingWriter = ac.FixedRolePrefix + "alerting:writer"
+
+	alertingReaderRole = ac.RoleRegistration{
+		Role: ac.RoleDTO{
+			Name:        alertingReader,
+			DisplayName: "Alerting Rules Reader",
+			Description: "Read alerting rules",
+			Group:       "Alerting",
+			Version:     1,
+			Permissions: []ac.Permission{
+				{
+					Action: ActionAlertingRuleRead,
+					Scope:  ScopeNamespaceAll,
+				},
+				{
+					Action: ActionAlertingRuleRead,
+					Scope:  ScopeDatasourcesAll,
+				},
+				{
+					Action: ActionAlertingInstanceRead, // scope is the current organization
+				},
+				{
+					Action: ActionAlertingNotificationsRead, // scope is the current organization
+				},
+			},
+		},
+		Grants: []string{string(models.ROLE_VIEWER)},
+	}
+
+	alertingWriterRole = ac.RoleRegistration{
+		Role: ac.RoleDTO{
+			Name:        alertingWriter,
+			DisplayName: "Alerting Rules Writer",
+			Description: "Read and update alerting rules",
+			Group:       "Alerting",
+			Version:     1,
+			Permissions: ac.ConcatPermissions(alertingReaderRole.Role.Permissions, []ac.Permission{
+				{
+					Action: ActionAlertingRuleCreate,
+					Scope:  ScopeNamespaceAll,
+				},
+				{
+					Action: ActionAlertingRuleUpdate,
+					Scope:  ScopeNamespaceAll,
+				},
+				{
+					Action: ActionAlertingRuleDelete,
+					Scope:  ScopeNamespaceAll,
+				},
+				{
+					Action: ActionAlertingRuleCreate,
+					Scope:  ScopeDatasourcesAll,
+				},
+				{
+					Action: ActionAlertingRuleUpdate,
+					Scope:  ScopeDatasourcesAll,
+				},
+				{
+					Action: ActionAlertingRuleDelete,
+					Scope:  ScopeDatasourcesAll,
+				},
+				{
+					Action: ActionAlertingInstanceCreate, // scope is the current organization
+				},
+				{
+					Action: ActionAlertingInstanceUpdate, // scope is the current organization
+				},
+				{
+					Action: ActionAlertingNotificationsCreate, // scope is the current organization
+				},
+				{
+					Action: ActionAlertingNotificationsUpdate, // scope is the current organization
+				},
+				{
+					Action: ActionAlertingNotificationsDelete, // scope is the current organization
+				},
+			}),
+		},
+		Grants: []string{string(models.ROLE_EDITOR)},
+	}
+)
+
+// DeclareFixedRoles registers the fixed roles provided by the alerting module
+func (api *API) DeclareFixedRoles() error {
+	return api.AccessControl.DeclareFixedRoles(
```

Comment:
I think we can skip to register the fixed roles before they have any use. Otherwise they will show up in the ui :)

Reply:
The idea of the PR is to let UI team start working on the feature. What would you recommend in this case? Cut a feature branch?